### DISCUSSION TITLE
Puppeteer does not find Chrome image

### DISCUSSION THREAD

**Discussion by enzoferey** [vercel/community#3074] [Help] [answered]
### Summary

Around 2-3 hours ago, after uploading a new build of a Next.js application that uses [`puppeteer`](https://pptr.dev/) inside one of its [API Route Handlers](https://nextjs.org/docs/app/building-your-application/routing/router-handlers), we have started seeing the following errors:

```
Error: Could not find Chrome (ver. 114.0.5735.133). This can occur if either
 1. you did not perform an installation before running the script (e.g. `npm install`) or
 2. your cache path is incorrectly configured (which is: /home/sbx_user1051/.cache/puppeteer).
For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration.
```

There have been many issues with Chromium images and Puppeteer in serverless environments, a quick Google search can confirm it. However, we are using the [new headless mode](https://developer.chrome.com/articles/new-headless/) so we were expecting the behavior might be a bit different and indeed it was. In fact, it has been working until 2-3 hours ago when we uploaded a new build (which is just some software change on other files).

We have tried setting `VERCEL_FORCE_NO_BUILD_CACHE` so the cache does not mess somehow with the puppeteer's one, but it did not solve the issue.

Do you have any pointers to fix the issue? Is this an update on the Vercel infra that has changed something about it?

Thank you!

### Steps to Reproduce

The dependencies at play:

"next": "13.4.7"
"puppeteer": "20.7.4"

**Comment by JosefHobler** [comment]
same problem here :/

**Reply by zacBkh** [reply]
Same issue... !

**Comment by Trustrecap** [comment]
Any update on this?

**Comment by Jamesllllllllll** [comment, marked as answer]
I ran into this issue and figured out that the Chromium binary exceeded the 50mb serverless function size.

I had success following this article: https://www.stefanjudis.com/blog/how-to-use-headless-chrome-in-serverless-functions/

**Reply by sKopheK** [reply]
still won't work on vercel and in many cases this can't be sustainable

**Reply by kasperaamodt** [reply]
This works like a charm, thank you!

**Reply by naseroish** [reply]
if any one getting 
[Error: ENOENT: no such file or directory, open '/tmp/chromium-pack/fonts.tar.br'] { errno: -2, code: 'ENOENT', syscall: 'open', path: '/tmp/chromium-pack/fonts.tar.br' }  
Try using the latest version of the cdn used u can find it here [https://github.com/Sparticuz/chromium/releases](https://github.com/Sparticuz/chromium/releases) 

then change the version in your code
example :
before
https://github.com/Sparticuz/chromium/releases/download/v116.0.0/chromium-v116.0.0-pack.tar
after
https://github.com/Sparticuz/chromium/releases/download/v123.0.1/chromium-v123.0.1-pack.tar

**Comment by Joead** [comment]
Having the same issue.

**Comment by therbta** [comment]
Having the same issue here.

```
"Could not find Chrome (ver. 123.0.6312.58). This can occur if either
 1. you did not perform an installation before running the script (e.g. `npx puppeteer browsers install chrome`) or
 2. your cache path is incorrectly configured (which is: ./.cache/puppeteer).
For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration."
```

**Comment by devtanna** [comment]
Having the same issue here :(
